zebra-network/src/peer_set/candidate_set/tests/vectors.rs

```diff
@@ -115,6 +130,77 @@ fn rejects_all_addresses_if_applying_offset_causes_an_underflow() {
     assert!(validated_peers.next().is_none());
 }
 
+/// Test that calls to [`CandidateSet::update`] are rate limited.
+#[test]
+fn candidate_set_updates_are_rate_limited() {
+    let runtime = Runtime::new().expect("Failed to create Tokio runtime");
+    let _guard = runtime.enter();
+
+    let address_book = AddressBook::new(&Config::default(), Span::none());
+    let mut candidate_set =
+        CandidateSet::new(Arc::new(Mutex::new(address_book)), mock_peer_service());
+
+    runtime.block_on(async move {
+        time::pause();
+
+        // Run the test for enough time for `update` to actually run three times
+        let time_limit =
+            Instant::now() + 3 * MIN_PEER_GET_ADDR_INTERVAL + StdDuration::from_secs(1);
```

Comment:
Nice catch, rounding might have got you just under 3 calls.

Also let's turn this 3 into a named constant, so it's clearly different from the other 3.